Add trace information to network request panel

ccschmitz opened the pull request:
## Summary

The main update in this PR is adding the trace flame graph and span details on the network resource panel. There are a number of other updates to traces and the flame graph added to support this.

* Moves the data to display a trace into a `TraceContext` which provides more flexibility for building UI with the trace data (e.g. allowed me to display span attributes and a flame graph separately on the network request panel).
* Fixes an issue with the time display of ticks for longer durations + updates the numbers to use scientific notation, which is easier to parse.
* Assigns `startTime` to all spans on the server which is the unix time in nanoseconds. This makes our flame graph span placement more precise since we could only get to millisecond precision when parsing `timestamp` in the client.
* Hides `children` in the span attributes, fixing some UI jank when updating the span attributes in the JSON viewer.
* Throttles the zoom handler and removes the scaling factor which makes it easier/faster to zoom into the area of the graph you want to see.

https://www.loom.com/share/144c34193ea149c4a309b3b817860bae

## How did you test this change?

Local click test. Can't be click tested in the PR previews because of the backend changes.

## Are there any deployment considerations?

Only backend change is adding `startTime` on spans. This should be fairly safe, but will click test to confirm as this rolls out.

## Does this work require review from our design team?

N/A - no design changes in this PR.

SpennyNDaJets reviewed (approved):
Code looks good to me! Just wanted to call out a couple things I noticed from click testing, but not sure if this is a local setup error or due to previous data formats?

1. Hit a type error when attempting to scroll - did not happen everytime though
2. "Invalid date" string in trace sidebar

![Uploading Screenshot 2023-10-25 at 12.55.38 PM.png…]()
![Screenshot 2023-10-25 at 12 56 34 PM](https://github.com/highlight/highlight/assets/17744174/6edcaa15-9c8e-4b14-a420-125b1d1b4d0d)

ccschmitz commented:
@SpennyNDaJets thank you for catching those issues! I addressed both and cleaned up a bunch of UI to better match [the latest mocks](https://www.figma.com/file/rdBPNzn7Klk6RG1LHUCE5B/Screen-Designs?type=design&node-id=12576%3A20863&mode=design&t=1WsLdq4ODD2Boskj-1) (added prev/next buttons + hotkeys).

<img width="696" alt="Screenshot 2023-10-25 at 3 03 40 PM" src="https://github.com/highlight/highlight/assets/308182/c784f176-f71e-4b56-901d-27191b989851">

SpennyNDaJets reviewed (approved):
![image](https://github.com/highlight/highlight/assets/17744174/99e43341-376a-40bb-8d4c-3106e66f260e)
